src/EditorFeatures/Core/Extensibility/NavigationBar/NavigationBarItems/RoslynNavigationBarItem.AbstractGenerateCodeItem.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Extensibility.NavigationBar
+{
+    internal abstract partial class RoslynNavigationBarItem
+    {
+        public abstract class AbstractGenerateCodeItem : RoslynNavigationBarItem
```

Comment:
these data types moved up from VB.  tehy moved to be pure data.  All logic in tehm moved into the INavigationBarItemService.